source/common/router/router.cc

```diff
@@ -1329,20 +1330,50 @@ Filter::UpstreamRequest::UpstreamRequest(Filter& parent, Http::ConnectionPool::I
       create_per_try_timeout_on_request_complete_(false) {
 
   if (parent_.config_.start_child_span_) {
-    span_ = parent_.callbacks_->activeSpan().spawnChild(
-        parent_.callbacks_->tracingConfig(), "router " + parent.cluster_->name() + " egress",
-        parent.timeSource().systemTime());
+
+    // Create span name based on attempt number.
+    std::string span_base_name = "router " + parent.cluster_->name() + " egress";
+    std::string span_name_with_retry =
```

Comment:
I don't think the span name should be changed, but instead the retry count added as a numeric tag.

Reply:
There is no tag for it, so I added one in `http_tracer_impl.h`. Let me know if this is OK.